File: common/src/main/java/net/impleri/playerskills/client/NetHandler.java
package net.impleri.playerskills.client;

import net.impleri.playerskills.PlayerSkills;
import net.impleri.playerskills.network.ResyncSkillsMessage;
import net.minecraft.client.Minecraft;

abstract class NetHandler {
    public static void resyncPlayer() {
        var player = Minecraft.getInstance().player;

        if (player == null) {
            return;
        }

        PlayerSkills.LOGGER.debug("Requesting skills resync for {}", player.getName().getString());

        new ResyncSkillsMessage(player).sendToServer();
    }
}

File: common/src/main/kotlin/net/impleri/playerskills/client/NetHandler.kt
package net.impleri.playerskills.client

import net.impleri.playerskills.network.ResyncSkillsMessage
import net.impleri.playerskills.utils.PlayerSkillsLogger
import net.minecraft.client.Minecraft

internal object NetHandler {
  fun resyncPlayer() {
    val player = Minecraft.getInstance().player ?: return
    PlayerSkillsLogger.SKILLS.debug("Requesting skills resync for ${player.name}")
    ResyncSkillsMessage(player).sendToServer()
  }
}
